actions/approve_csr_handler.go

```diff
@@ -100,17 +95,19 @@ func (h *approveCSRHandler) handle(ctx context.Context, log logrus.FieldLogger,
 	return errors.New("certificate signing request was not approved")
 }
 
-func (h *approveCSRHandler) getInitialNodeCSR(ctx context.Context, nodeName string) (*csr.Certificate, error) {
-	csrFetchCtx, cancel := context.WithTimeout(ctx, h.initialCSRFetchTimeout)
+func (h *approveCSRHandler) getInitialNodeCSR(ctx context.Context, log *logrus.Entry, nodeName string) (*csr.Certificate, error) {
+	log.Debug("getting initial csr")
+
+	ctx, cancel := context.WithTimeout(ctx, h.initialCSRFetchTimeout)
 	defer cancel()
 
 	for {
 		select {
-		case <-csrFetchCtx.Done():
-			return nil, csrFetchCtx.Err()
+		case <-ctx.Done():
+			return nil, ctx.Err()
 		case <-time.After(h.csrFetchInterval):
 			cert, err := csr.GetCertificateByNodeName(ctx, h.clientset, nodeName)
-			if err != nil && !errors.Is(err, csr.ErrNodeCertificateNotFound) {
+			if err != nil && errors.Is(err, context.DeadlineExceeded) {
```

Comment:
```suggestion
			if errors.Is(err, context.DeadlineExceeded) {
```

Do we want to execit only on context deadline exceeded error?

Reply:
changed to retry other errors only few times